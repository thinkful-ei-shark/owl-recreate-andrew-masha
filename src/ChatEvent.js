import React from 'react';

function ChatEvent(props) {
  const eventTexts = {
    'thumbs-up': 'gave a thumbs up',
    'thumbs-down': 'gave a thumbs down',
    'raised-hand': 'raised their hand',
    'clap': 'clapped',
    'join': 'joined',
    'join-stage': 'joined the stage',
    'leave': 'left',
    'leave-stage': 'left the stage'
  };
  return (
    <li className={props.event.type}>
      <span className='participant-name'>{props.participant.name}</span>
      <span className='event-text'>{eventTexts[props.event.message]}</span>
    </li>
  )
}

export default ChatEvent;