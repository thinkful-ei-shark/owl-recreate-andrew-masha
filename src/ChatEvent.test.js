import React from 'react';
import ReactDOM from 'react-dom'

import ChatEvent from './ChatEvent.js'

const testProps = {
  participant:
  {
    id: 1,
    name: 'Koren Templeton',
    avatar:
      'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true
  },
  event: 
  {
    participantId: 3,
    type: 'message',
    message: 'Hello world',
    time: 1548852646559,
    timestamp: 1548852484247
  }
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <ChatEvent event={testProps.event} participant={testProps.participant} />,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
})