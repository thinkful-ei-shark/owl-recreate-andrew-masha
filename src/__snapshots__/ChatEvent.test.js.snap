// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<li
  className="message"
>
  <span
    className="participant-name"
  >
    Koren Templeton
  </span>
  <span
    className="event-text"
  />
</li>
`;
